substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/VM.java

```diff
@@ -36,28 +35,34 @@ public final class VM {
     public final String vendor;
     public final String vendorUrl;
     public final String vendorVersion;
-    public final String supportURL;
 
     @Platforms(Platform.HOSTED_ONLY.class)
     public VM(String vmInfo) {
         info = vmInfo;
-        supportURL = System.getProperty("org.graalvm.supporturl", "https://graalvm.org/native-image/error-report/");
-        version = stripJVMCISuffix(System.getProperty("java.runtime.version"));
+        version = getVersion();
         vendor = System.getProperty("org.graalvm.vendor", "GraalVM Community");
         vendorUrl = System.getProperty("org.graalvm.vendorurl", "https://www.graalvm.org/");
-        vendorVersion = System.getProperty("org.graalvm.vendorversion", "GraalVM CE");
+        vendorVersion = getVendorVersion();
     }
 
     @Platforms(Platform.HOSTED_ONLY.class)
-    public static VM getErrorReportingInstance() {
-        if (ImageSingletonsSupport.isInstalled() && ImageSingletons.contains(VM.class)) {
-            return ImageSingletons.lookup(VM.class);
+    public static String getSupportUrl() {
+        return System.getProperty("org.graalvm.supporturl", "https://graalvm.org/native-image/error-report/");
+    }
+
+    public static String getVendorVersion() {
+        if (ImageInfo.inImageRuntimeCode()) {
+            return System.getProperty("java.vendor.version");
         } else {
-            // create a fall back instance
-            return new VM(System.getProperty("java.vm.info", ""));
+            return System.getProperty("org.graalvm.vendorversion", "GraalVM CE");
         }
     }
 
+    @Platforms(Platform.HOSTED_ONLY.class)
+    public static String getVersion() {
+        return stripJVMCISuffix(System.getProperty("java.runtime.version"));
```

Comment:
I wonder if stripping the JVMCI version is a good idea. This results in users assuming they are running the equivalent of the upstream OpenJDK version while they are actually running a much different variant.

E.g. for ` labsjdk-ce-17.0.7+4-jvmci-23.0-b10` we see 206 commits and 331 files changed when [compared to the upstream `17.0.7+4` tag](https://github.com/graalvm/labs-openjdk-17/compare/jdk-17.0.7+4...jvmci-23.0-b10 ).

Reply:
SVM does not implement JVMCI, which is why we're stripping this here. The JVMCI suffix is still part of the `java.runtime.version` property though (see [here](https://github.com/oracle/graal/pull/6302#issue-1644267829)).

Reply:
It might not implement JVMCI, i.e. expose it to users, but still uses it during code generation so it's version can affect the produced native executable.

Reply:
Similar to the vendor version, this version is used during code generation and for a java.* property within an executable. I guess we could do the same dance we're doing with the vendor version, not strip during code generation and strip for the property. Note that the value is derived from `java.runtime.version`, which is displayed as part of `native-image --version`.
Are you suggesting to show `java.runtime.version` in the build output as well?

Reply:
Since this is showing up as `Java version: XXX` in the build output and we have a `vendor version: YYY` just after it, my take would be that it should be in the vendor version, i.e. `YYY`, info. If at all. No strong feelings about it.

Reply:
What I strongly suggest is to keep using the internal version info (`23.1.0` for master, `23.0.0` for release 23) of the graal codebase in the Graal VM CE *vendor* version. There is no point in repeating the JDK version there. At least until svm in its entirety is part of OpenJDK ;-) After all it's a separate code base as of yet. Having users needing to dig into `release` file for that seems inconvenient.

Reply:
The point of not including the GraalVM version in the vendor version is to be able to phase it out. We don't need it anymore because we adopt the OpenJDK versioning scheme as part of the alignment. Unfortunately, we can't update all references in our code base at the same time, so we're prioritizing the user-facing bits at the moment. The upcoming release includes JDK17 and JDK20 builds, both sourced from the same 23.0 code base. We and you know about this, but users no longer need to see it anymore.

Reply:
I get that you want to phase it out. That seems fine. The vendor version shouldn't be used anywhere by users. I think the majority gets that as shown below in some examples. But some sort of acknowledgement needs to be there that it's not just OpenJDK. I.e. version the code base itself. Whether or not it's `17.0.7+4.1` or `23.0.1` in the vendor version makes no difference for the user. Both are similarly meaningless to most but expert users. Yet, the former makes it unnecessarily obscure to figure out the actual code base in use and outright contains bits which doesn't give any clue about the version in use (`17.0.7+4` comes from somewhere else, not relating to the svm codebase and is already shown plenty in other bits of the version output). The beauty of the vendor version is that its precisely that, just some hints about the vendor version. It could get removed entirely once svm is actually in the OpenJDK code base. But we aren't there yet. I'd assume it would make life of support folks harder on your end as well.

Just some examples of vendor versions of different OpenJDK builds all of which make sense. Note that those are actually OpenJDK code-bases so the analogy is somewhat skewed:

```
openjdk 20-beta 2023-03-21
OpenJDK Runtime Environment Temurin-20+36-202303010331 (build 20-beta+36-202303010331)
OpenJDK 64-Bit Server VM Temurin-20+36-202303010331 (build 20-beta+36-202303010331, mixed mode, sharing)
```

```
openjdk 19.0.2 2023-01-17
OpenJDK Runtime Environment (Red_Hat-19.0.2.0.7-2.rolling.fc37) (build 19.0.2+7)
OpenJDK 64-Bit Server VM (Red_Hat-19.0.2.0.7-2.rolling.fc37) (build 19.0.2+7, mixed mode, sharing)
```

```
openjdk 11.0.18 2023-01-17 LTS
OpenJDK Runtime Environment Microsoft-7208460 (build 11.0.18+10-LTS)
OpenJDK 64-Bit Server VM Microsoft-7208460 (build 11.0.18+10-LTS, mixed mode)
```

With that in mind, the old version output for OpenJDK with libgraal made the most sense:

```
openjdk 17.0.4 2022-07-19
OpenJDK Runtime Environment GraalVM CE 22.3.0-dev (build 17.0.4+7-jvmci-22.3-b02)
OpenJDK 64-Bit Server VM GraalVM CE 22.3.0-dev (build 17.0.4+7-jvmci-22.3-b02, mixed mode, sharing)
```